lib/agent0/agent0/hyperdrive/policies/lpandarb_test.py

```diff
@@ -300,3 +319,166 @@ def test_reduce_short(interactive_hyperdrive: InteractiveHyperdrive, arbitrage_a
+    # use rust to predict trade outcome
+    bonds_after_fees = interactive_hyperdrive.hyperdrive_interface.calc_open_long(base_needed)
+    logging.info("bonds_after_fees is %s", bonds_after_fees)
+    bond_fees = bonds_after_fees * price_discount * curve_fee
+    bond_fees_to_pool = bond_fees * (FixedPoint(1) - governance_fee)
+    bond_fees_to_gov = bond_fees * governance_fee
+    bonds_before_fees = bonds_after_fees + bond_fees_to_pool + bond_fees_to_gov
+    logging.info("bonds_before_fees is %s", bonds_before_fees)
+    logging.info("bond_fees_to_pool is %s", bond_fees_to_pool)
+    logging.info("bond_fees_to_gov is %s", bond_fees_to_gov)
+
+    predicted_delta_bonds = -bonds_after_fees - bond_fees_to_gov
+    predicted_delta_shares = base_needed / share_price * (FixedPoint(1) - price_discount * curve_fee * governance_fee)
+    logging.info("predicted delta bonds is %s", predicted_delta_bonds)
+    logging.info("predicted delta shares is %s", predicted_delta_shares)
+
+    # measure pool before trade
+    pool_state_before = deepcopy(interactive_hyperdrive.hyperdrive_interface.current_pool_state)
+    pool_bonds_before = pool_state_before.pool_info.bond_reserves
+    pool_shares_before = pool_state_before.pool_info.share_reserves
+    # do the trade
+    event = arbitrage_andy.open_long(base=base_needed)
+    event = event[0] if isinstance(event, list) else event
+    actual_event_bonds = event.bond_amount
+    actual_event_base = event.base_amount
+    logging.info(
+        "opened long with input base=%s, output Δbonds= %s%s, Δbase= %s%s",
+        base_needed,
+        "+" if actual_event_bonds > 0 else "",
+        actual_event_bonds,
+        "+" if actual_event_base > 0 else "",
+        actual_event_base,
+    )
+    # measure pool after trade
+    pool_state_after = deepcopy(interactive_hyperdrive.hyperdrive_interface.current_pool_state)
+    pool_bonds_after = pool_state_after.pool_info.bond_reserves
+    pool_shares_after = pool_state_after.pool_info.share_reserves
+    logging.info("pool bonds after is %s", pool_bonds_after)
+    logging.info("pool shares after is %s", pool_shares_after)
+    actual_delta_bonds = pool_bonds_after - pool_bonds_before
+    actual_delta_shares = pool_shares_after - pool_shares_before
+    logging.info("actual delta bonds is %s", actual_delta_bonds)
+    logging.info("actual delta shares is %s", actual_delta_shares)
+
+    bonds_discrepancy = float((actual_delta_bonds - predicted_delta_bonds) / predicted_delta_bonds)
+    shares_discrepancy = float((actual_delta_shares - predicted_delta_shares) / predicted_delta_shares)
+    logging.info(f"discrepancy (%) for bonds is {bonds_discrepancy:e}")  # pylint: disable=logging-fstring-interpolation
+    logging.info(
+        f"discrepancy (%) for shares is {shares_discrepancy:e}"
+    )  # pylint: disable=logging-fstring-interpolation
+
+    assert abs(bonds_discrepancy) < 1e-7
+    assert abs(shares_discrepancy) < 1e-7
```

Comment:
We do have an `fp_isclose` function [here](https://github.com/delvtech/agent0/blob/e8a8cfca8a622e65d7e29ac70e4f4c920acf5bab/lib/agent0/agent0/interactive_fuzz/helpers/fuzz_assertion_exception.py#L9). It is in a inconvenient location for this test, since so far it has only been used by fuzz tests. I made an issue for switching to this: https://github.com/delvtech/agent0/issues/1286